controllers/vspherecluster_controller.go

```diff
@@ -194,6 +194,16 @@ func (r clusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr er
 func (r clusterReconciler) reconcileDelete(ctx *context.ClusterContext) (reconcile.Result, error) {
 	ctx.Logger.Info("Reconciling VSphereCluster delete")
 
+	vsphereMachines, err := infrautilv1.GetVSphereMachinesInCluster(ctx, ctx.Client, ctx.VSphereCluster.Namespace, ctx.VSphereCluster.Name)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err,
```

Comment:
Suggestion (possibly for follow-up): emit an event indicating deletion failed & include the error message.

Reply:
I plan to raise a ticket for Eventing in CAPV, so will address this in a follow-up

Reply:
same